src/protocol/prss.rs

```diff
@@ -85,7 +88,9 @@ impl IndexedSharedRandomness {
         {
             self.used.insert(index);
         }
-
+        if let Some(ref sink) = self.metrics_sink {
+            sink.clone().increment_counter(INDEXED_PRSS_GENERATED, 1);
```

Comment:
why clone?

Reply:
the design of the `Sink` is a bit weird, `increment_counter` requires an exclusive reference, despite the fact that it claims to be cheaply cloneable